bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTHandler.java

```diff
@@ -0,0 +1,278 @@
+
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+
+import net.heberling.ismart.abrp.ABRP;
+import net.heberling.ismart.asn1.v1_1.entity.Message;
+import net.heberling.ismart.asn1.v2_1.MessageCoder;
+import net.heberling.ismart.asn1.v2_1.entity.OTA_RVCReq;
+import net.heberling.ismart.asn1.v2_1.entity.OTA_RVCStatus25857;
+import net.heberling.ismart.asn1.v2_1.entity.OTA_RVMVehicleStatusResp25857;
+import net.heberling.ismart.asn1.v2_1.entity.RvcReqParam;
+import net.heberling.ismart.asn1.v3_0.entity.OTA_ChrgMangDataResp;
+
+/**
+ * The {@link SAICiSMARTHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+public class SAICiSMARTHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    @Nullable
+    SAICiSMARTVehicleConfiguration config;
+    private @Nullable Future<?> pollingJob;
+    private ZonedDateTime lastAlarmMessage;
+    private ZonedDateTime lastCarActivity;
+
+    /**
+     * If the binding is initialized, treat the car as active (lastCarActivity = now) to get some first data.
+     * 
+     * @param httpClientFactory
+     * @param timeZoneProvider
+     * @param thing
+     */
+    public SAICiSMARTHandler(TimeZoneProvider timeZoneProvider, Thing thing) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        lastAlarmMessage = ZonedDateTime.now(getTimeZone());
+        lastCarActivity = ZonedDateTime.now(getTimeZone());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(SAICiSMARTBindingConstants.CHANNEL_FORCE_REFRESH) && command == OnOffType.ON) {
```

Comment:
A switch might be more readable then all the if statements, espacially when this binding grows in the future.  For your consideration.

Reply:
good idea I can refactor this if I change it in the future.